plugins/plugin_manager.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ Copyright 2025 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+#include <iostream>
+#include <string>
+#include "plugin_manager.h"
+
+using CreatePluginFunc = IDivePlugin* (*)();
+
+PluginManager::PluginManager(MainWindow& main_window) :
+    m_main_window(main_window),
+    m_library_loader(CreateDynamicLibraryLoader())
+{
+}
+
+PluginManager::~PluginManager()
+{
+    UnloadPlugins();
+}
+
+void PluginManager::LoadPlugins(const std::filesystem::path& plugins_dir_path)
+{
+    std::string expected_suffix = m_library_loader->GetPluginFileExtension();
+
+    for (const auto& entry : std::filesystem::directory_iterator(plugins_dir_path))
+    {
+        if (!entry.is_regular_file())
+        {
+            continue;
+        }
+
+        std::string file_name = entry.path().filename().string();
+        std::string file_path = entry.path().string();
+
+        auto HasSuffix = [&](const std::string& file_name, const std::string& expected_suffix) {
+            return file_name.length() < expected_suffix.length() ||
+                   file_name.substr(file_name.length() - expected_suffix.length()) !=
+                   expected_suffix;
+        };
+
+        if (HasSuffix(file_name, expected_suffix))
```

Comment:
Oh I just found out that std::filesystem::path has an `extension()` function, can we use that here instead?

```suggestion
        if (entry.path().extension() == m_library_loader->GetPluginFileExtension())
```

Reply:
ah, thanks! make sense